.github/workflows/publish.yml

```diff
@@ -43,6 +43,29 @@ jobs:
                   name: release
                   path: dist/
 
+    pre-commit:
```

Comment:
Why would this be necessary? The pre-commit test is run by [pre-commit.ci](https://pre-commit.ci/), so that seems like a redundant thing?

Reply:
It's needed, because to run mypy correctly we need to install all the dependencies, and `pre-commit.ci` doesn't allow that. We have the same problem in aiida-core, where we recently removed this "redundant" pre-commit job without understanding the ramifications. I'll open an issue there with more details.